trl/trainer/dpo_trainer.py

```diff
@@ -97,6 +97,8 @@ def __init__(
         preprocess_logits_for_metrics: Optional[Callable[[torch.Tensor, torch.Tensor], torch.Tensor]] = None,
         max_length: Optional[int] = None,
         max_prompt_length: Optional[int] = None,
+        max_target_length: Optional[int] = None,
+        is_encoder_decoder: bool = None,
```

Comment:
```suggestion
```

Reply:
You can use `model.config.is_encoder_decoder`:
https://github.com/huggingface/transformers/blob/main/src/transformers/configuration_utils.py#L277-L278
All Configs have this option.

Do note however that in this `__init__` we might have a `model_init` instead of a `model`, so we'll have to work around that.

Reply:
Oh yes thanks I forget configs had this option